internal/action/render.go

```diff
@@ -140,6 +150,35 @@ func (r Render) imageToDeclcfg(ctx context.Context, imageRef string) (*declcfg.D
 	return cfg, nil
 }
 
+// checkDBFile returns an error if ref is not an sqlite3 database.
+func checkDBFile(ref string) error {
+	f, err := os.Open(ref)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if cerr := f.Close(); cerr != nil {
+			logrus.Error(cerr)
+		}
+	}()
+
+	// Just read enough bytes to detect if ref is an sqlite3 db
+	// so the whole file is not loaded into memory.
+	var buf [4]byte
+	if _, err := f.Read(buf[:]); err != nil {
+		return fmt.Errorf("error reading first 4 bytes of ref %q: %v", ref, err)
+	}
+
+	typ, err := filetype.Match(buf[:])
```

Comment:
Use `filetype.MatchFile()` here instead?

Reply:
:+1: `MatchFile()` only [loads 8kB](https://github.com/h2non/filetype/blob/master/match.go#L56) into memory which is fine.